app/modules/tmr.c

```diff
@@ -135,20 +141,32 @@ static int tmr_now(lua_State* L){
 	return 1; 
 }
 
-// Lua: tmr.register( id, interval, mode, function )
+static int tmr_get( lua_State *L, int stack, timer_t *tmr ) {
+  if (lua_isuserdata(L, stack)) {
+    *tmr = (timer_t)luaL_checkudata(L, stack, "tmr.timer");
+    if (!*tmr) return luaL_error(L, "invalid userdata");
```

Comment:
No need to check `tmr` for != NULL. `luaL_checkudata()` will error out if the metatable doesn't match AFAIK.